adhoc/us-supreme-court-cases/import-transcripts.py

```diff
@@ -0,0 +1,154 @@
+#!/usr/bin/python3
+
+import datetime
+import json
+import pandas
+import glob
+import re
+import unidecode
+import copy
+
+from doltpy.core import Dolt
+from pprint import pprint
+
+# missing some here
+case_column_map = {
+    'term' : 'term',
+    'caseName': 'case_name',
+    'caseLink': 'link', 
+    'decidedBy': 'decided_by',
+    'arguedOn': 'argued_on', 
+    'petitioner': 'petitioner',
+    'respondent': 'respondent',
+    'docket': 'docket',
+    'citation': 'citation',
+    'justiaLink': 'justia_link',
+    'lowerCourt': 'lower_court',
+    'grantedOn': 'granted_on',
+    'decidedOn': 'decided_on',
+    'wikiData': {
+        'title': 'wiki_title',
+        'Litigants': 'wiki_litigants',
+        'ArgueDate': 'wiki_argue_date',
+        'ArgueYear': 'wiki_argue_year',
+        'DecideDate': 'wiki_decide_date',
+        'DecideYear': 'wiki_decide_year',
+        'FullName': 'wiki_full_name',
+        'USVol': 'wiki_us_vol',
+        'USPage': 'wiki_us_page',
+        'ParallelCitations': 'wiki_parallel_citations',
+        'Docket': 'wiki_docket',
+        'OralArgument': 'wiki_oral_argument',
+        'OralReargument': 'wiki_oral_reargument',
+        'OpinionAnnouncement': 'wiki_opinion_announcement',
+        'Prior': 'wiki_prior',
+        'Subsequent': 'wiki_subsequent',
+        'Holding': 'wiki_holding',
+        'SCOTUS': 'wiki_scotus',
+        'Majority': 'wiki_majority',
+        'JoinMajority': 'wiki_join_majority',
+        'Plurality': 'wiki_plurality',
+        'JoinPlurality': 'wiki_join_plurality',
+        'Concurrence': 'wiki_concurrence',
+        'JoinConcurrence': 'wiki_join_concurrence',
+        'Concurrence2': 'wiki_concurrence2',
+        'JoinConcurrence2': 'wiki_join_concurrence2',
+        'Concurrence3': 'wiki_concurrence3',
+        'Concurrence/Dissent': 'wiki_concurrence_dissent',
+        'JoinConcurrence/Dissent': 'wiki_join_concurrence_dissent',
+        'Dissent': 'wiki_dissent',
+        'JoinDissent': 'wiki_join_dissent',
+        'Dissent2': 'wiki_dissent2',
+        'JoinDissent2': 'wiki_join_dissent2',
+        'NotParticipating': 'wiki_not_participating',
+        'LawsApplied': 'wiki_laws_applied'
+    }
+}
+
+transcript_column_map = {
+    'transcriptTitle': 'title',
+    'transcriptLink': 'link',
+}
+
+text_column_map = {
+    'start': 'start',
+    'stop': 'stop',
+    'duration': 'duration',
+    'text': 'text',
+}
+
+date_convert = [
+    'argued_on',
+    'granted_on',
+    'decided_on'
+]
+
+def normalize(data, mapper, output):
+    for key in dict.keys(mapper):
+        if key in data:
+            if isinstance(mapper[key], dict):
+                normalize(data[key], mapper[key], output)
+            else:
+                # Strip unicode
+                if isinstance(data[key], str):
+                    output[mapper[key]] = unidecode.unidecode(data[key])
+                else:
+                    output[mapper[key]] = data[key]
+
+def import_case_file(path, case, transcripts):
```

Comment:
You shouldn't mutate objects that are passed as parameters. Instead construct them in the function body and return them as values. They will, in a technical sense, be passed by reference.

Reply:
Point me at an example. I can return an object for cases but transcripts gets appended to every time. I need to edit that one in place or return an array and join it to the end of the other one outside the function. That seems lame.